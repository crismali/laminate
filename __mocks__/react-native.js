/* eslint-env jest */

class Value {
  constructor (value) {
    this.value = value
  }

  interpolate (args) {
    return { interpolated: this.value, args: args }
  }
}

const View = ({ children }) => children
View.displayName = 'Animated.View'

const Animated = { Value, View }

const Dimensions = { get: jest.fn(() => ({ width: 375, height: 667 })) }

const StyleSheet = { create: jest.fn(styles => styles) }

export {
  Animated,
  Dimensions,
  StyleSheet
}
