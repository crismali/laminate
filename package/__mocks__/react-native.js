/* eslint-env jest */
import React from 'react'

class Value {
  constructor (value) {
    this.value = value
  }

  interpolate (args) {
    return { interpolated: this.value, args: args }
  }

  setValue (value) {
    this.value = value
  }
}

const AnimatedView = ({ children }) => children
AnimatedView.displayName = 'Animated.View'

const animatedTimingStart = jest.fn((func) => func && func())
const Animated = {
  Value,
  View: AnimatedView,
  timing: jest.fn(() => ({ start: animatedTimingStart })),
  animatedTimingStart
}

const Dimensions = { get: jest.fn(() => ({ width: 375, height: 667 })) }

const Easing = { inOut: jest.fn() }

const PanResponder = { create: jest.fn((args) => ({ panHandlers: args })) }

const View = ({ children }) => <div>{children}</div> // eslint-disable-line

const StyleSheet = { create: jest.fn(styles => styles) }

export {
  Animated,
  Dimensions,
  Easing,
  PanResponder,
  StyleSheet,
  View
}
