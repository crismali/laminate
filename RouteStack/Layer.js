import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Animated, Dimensions, StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    position: 'absolute',
    width: '100%',
    height: '100%',
    top: 0,
    left: 0
  }
})

export class Layer extends Component {
  static propTypes = {
    children: PropTypes.any.isRequired,
    index: PropTypes.number.isRequired,
    animation: PropTypes.instanceOf(Animated.Value).isRequired
  }

  render () {
    const { children, animation, index } = this.props
    const { width } = Dimensions.get('window')

    const slide = animation.interpolate({
      inputRange: [0, 1],
      outputRange: [width, 0]
    })

    return <Animated.View style={[styles.container, { zIndex: index, left: slide }]}>
      {children}
    </Animated.View>
  }
}
