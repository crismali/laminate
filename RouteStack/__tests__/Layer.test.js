/* eslint-env jest */
import React from 'react'
import { Animated } from 'react-native'
import { shallow } from 'enzyme'
import { Layer } from '../Layer'

it('renders its children with a slide animation', () => {
  const value = new Animated.Value(0)
  const Child = () => 'some child'
  const subject = shallow(<Layer animation={value} index={123}>
    <Child />
  </Layer>)

  expect(subject).toMatchSnapshot()
})
