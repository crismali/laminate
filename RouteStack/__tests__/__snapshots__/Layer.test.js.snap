// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders its children with a slide animation 1`] = `
<Animated.View
  style={
    Array [
      Object {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      },
      Object {
        "left": Object {
          "args": Object {
            "inputRange": Array [
              0,
              1,
            ],
            "outputRange": Array [
              375,
              0,
            ],
          },
          "interpolated": 0,
        },
        "zIndex": 123,
      },
    ]
  }
>
  <Child />
</Animated.View>
`;
